src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/services/CFTSensitiveTaskEventLogsDatabaseService.java

```diff
@@ -37,12 +39,13 @@ public CFTSensitiveTaskEventLogsDatabaseService(SensitiveTaskEventLogsRepository
     private SensitiveTaskEventLog saveSensitiveTaskEventLog(SensitiveTaskEventLog sensitiveTaskEventLog) {
         try {
             return sensitiveTaskEventLogsRepository.save(sensitiveTaskEventLog);
-        } catch (Error e) {
-            log.info("Couldn't save SensitiveTaskEventLog");
+        } catch (IllegalArgumentException e) {
+            log.error("Couldn't save SensitiveTaskEventLog for taskId {}", sensitiveTaskEventLog.getTaskId());
             return sensitiveTaskEventLog;
         }
     }
 
+    @Transactional(propagation = Propagation.REQUIRES_NEW)
```

Comment:
Is this still a nested transaction? I.e. you start the primary tx and then _during_ the execution of that tx you open a new one?  If so then I suspect the first tx gets suspended and waits for the 2nd one to complete before completing itself?
See this: https://docs.spring.io/spring-framework/docs/4.2.x/spring-framework-reference/html/transaction.html#tx-propagation

Reply:
So what we really want is for the first tx to complete and only then start a new tx.  Just avoid a nested tx altogether?

Reply:
As per Slack, what you have works so we have the option of moving it forward and having a separate ticket to look at the transaction aspect as an NFR.  If you create a ticket and send it to me we can get this one approved.